plugins/webhooks/helper.js

```diff
@@ -365,7 +364,43 @@ async function triggeredPipelines(
 
     const currentRepoPipelines = pipelinesOnCommitBranch.concat(pipelinesOnOtherBranch);
 
-    return currentRepoPipelines.concat(pipelinesWithSubscribedRepos);
+    if (pipelinesOnCommitBranch.length === 0) {
+        return currentRepoPipelines;
+    }
+
+    // process the pipelinesWithSubscribedRepos only when the pipelinesOnCommitBranch is not empty
+    // pipelinesOnCommitBranch has the information to determine the triggering event of downstream subscribing repo
+    pipelinesWithSubscribedRepos.forEach(p => {
+        if (!Array.isArray(p.subscribedScmUrlsWithActions)) {
+            return;
+        }
+        p.subscribedScmUrlsWithActions.forEach(subscribedScmUriWithAction => {
+            const { scmUri: subscribedScmUri, actions: subscribedActions } = subscribedScmUriWithAction;
+
+            if (pipelinesOnCommitBranch[0].scmUri === subscribedScmUri) {
```

Comment:
should check length of arr when indexing

Reply:
these lines should be sufficient to ensure that `pipelinesOnCommitBranch` at least is a length of 1. 
https://github.com/screwdriver-cd/screwdriver/blob/c3d7ce4a9a22a2e98233fcf85c3245d774f4ede2/plugins/webhooks/helper.js#L367-L369

Reply:
and I have also added a comment that the subsequent block will only proceed when `pipelinesOnCommitBranch` is a length of at least 1. 

https://github.com/screwdriver-cd/screwdriver/blob/c3d7ce4a9a22a2e98233fcf85c3245d774f4ede2/plugins/webhooks/helper.js#L371-L372